README.md

```diff
@@ -206,36 +205,59 @@ assert.isFalse(result.success, result.error);
 ```
 
 ## Contracts with State
-sCrypt offers [stateful contracts](https://medium.com/xiaohuiliu/stateful-smart-contracts-on-bitcoin-sv-c24f83a0f783). `OP_RETURN` data of the contract locking script can be accessed by using an accessor named `dataPart`, for example:
+sCrypt offers [stateful contracts](https://scryptdoc.readthedocs.io/en/latest/state.html#stateful-contract). Declare any property that is part of the state with a decorator `@state` in a contract, for example:
 ```typescript
-const dataPart = instance.dataPart;
-const dataPartASM = instance.dataPart.toASM();
-const dataPartHex = instance.dataPart.toHex();
-// to set it using ASM
-instance.setDataPart(dataInASM);
-// to set it using state object (no nesting)
-let state = {'counter': 11, 'bytes': '1234', 'flag': true}
-instance.setDataPart(state)
+contract Counter {
+    @state
+    int counter;
+
+    constructor(int counter) {
+        this.counter = counter;
+    }
+}
 ```
-After that, the `instance.lockingScript` would include the data part automatically.
 
-If you want to access the code part of the contract's locking script including the trailing `OP_RETURN`, use:
+Use the initial state to instantiate the contract and read the state by accessing the properties of the contract instance.
+
 ```typescript
-const codePart = instance.codePart;
-const codePartASM = instance.codePart.toASM();
-const codePartHex = instance.codePart.toHex();
+const instance = new Counter(0);
+
+let state = instance.counter;
+```
+
+Then use `counter.getStateScript` to get a locking script include the new state automatically and use `counter.lockingScript` to get Sighash Preimage of the transaction.  
```

Comment:
Then use `counter.getStateScript()` to get a locking script that includes the new state.

Reply:
Also add a detailed comment on what the input/parameter of the function is, in what format. This is an important function.